staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -184,10 +180,42 @@ func (p *Preferences) applyOverrides(rootCmd *cobra.Command, kuberc *config.Pref
 // of the command. Lastly, others parameters (e.g. resources, etc.) that are passed as arguments in kuberc
 // is appended into the command args.
 func (p *Preferences) applyAliases(rootCmd *cobra.Command, kuberc *config.Preference, args []string, errOut io.Writer) ([]string, error) {
-	_, _, err := rootCmd.Find(args[1:])
+	// First, always register all aliases as commands for help display
+	for _, alias := range kuberc.Aliases {
+		p.aliases[alias.Name] = struct{}{}
+
+		// Check if alias already exists or conflicts with built-in command
+		if _, _, err := rootCmd.Find([]string{alias.Name}); err == nil {
+			continue
+		}
+
+		// Find the target command to ensure it exists
+		commands := strings.Fields(alias.Command)
+		existingCmd, _, err := rootCmd.Find(commands)
+		if err != nil {
+			continue
+		}
+
+		aliasCmd := &cobra.Command{
+			Use:    alias.Name,
+			Short:  buildFullAliasDescription(alias),
+			Run:    existingCmd.Run,
+			RunE:   existingCmd.RunE,
+			Hidden: false,
+		}
+
+		aliasCmd.Flags().AddFlagSet(existingCmd.Flags())
+		aliasCmd.PersistentFlags().AddFlagSet(existingCmd.PersistentFlags())
```

Comment:
Please have a look at this comment https://github.com/kubernetes/kubernetes/pull/133050#discussion_r2287699129.

Reconstruction of aliases looks to me fragile and error-prone.